sled-agent/src/bootstrap/server.rs

```diff
@@ -108,14 +123,31 @@ impl Server {
 struct Inner {
     listener: TcpListener,
     sp: Option<SpHandle>,
+    // Note: This is a `std::sync::Mutex`, not a tokio Mutex. We only hold the
+    // lock briefly to either clone the list of certs or replace it; we never
+    // hold it across an `.await` (which wouldn't compile, so we know we're not
+    // doing so accidentally).
+    trust_quorum_members: Arc<Mutex<Option<Vec<Ed25519Certificate>>>>,
     bootstrap_agent: Arc<Agent>,
     log: Logger,
 }
 
 impl Inner {
     async fn start(
         bind_address: SocketAddrV6,
+        // TODO-cleanup `sp` is optional because we support running without an
+        // SP / any trust quorum mechanisms. Eventually it should be required.
         sp: Option<SpHandle>,
+        // `trust_quorum_members` should be `Some(_)` if we have already
+        // received a sled-agent request from RSS, but will be `None` prior to
+        // RSS completing rack setup. We're unable to verify that the sender of
+        // the sled-agent request is a member of our trust quorum, because it's
+        // that request that defines the trust quorum members. If
+        // `trust_quorum_members` is `None` and we receive a sled-agent request
+        // (and successfully handle it), we'll update
+        // `self.trust_quorum_members` to remember the members sent as part of
+        // that request.
+        trust_quorum_members: Option<Vec<Ed25519Certificate>>,
```

Comment:
> we will end up having code eventually where we read the trust quorum state if it exists off disk rather than waiting for RSS, which will not be active in the common case.

This should already be in place: `Agent` will [read an existing sled agent request](https://github.com/oxidecomputer/omicron/blob/af15a21377ec35fde9bd38981b5455b333278f56/sled-agent/src/bootstrap/agent.rs#L176-L191) if we've previously received one from RSS, which will then be provided to `Inner` [when it's created](https://github.com/oxidecomputer/omicron/blob/af15a21377ec35fde9bd38981b5455b333278f56/sled-agent/src/bootstrap/server.rs#L78-L81).

I do like the suggestion of switching from `Option` to an explicit enum, though. I'm going to go ahead and do that and fix the race I mentioned in https://github.com/oxidecomputer/omicron/pull/1191/files/af15a21377ec35fde9bd38981b5455b333278f56#r895841083.

Reply:
Thanks for pointing this out!